gprofiler/profilers/perf.py

```diff
@@ -203,14 +205,18 @@ def __init__(
             self._perfs.append(self._perf_dwarf)
         else:
             self._perf_dwarf = None
-
+        self.perf_node_attach = perf_node_attach
         assert self._perf_fp is not None or self._perf_dwarf is not None
 
     def start(self) -> None:
+        if self.perf_node_attach:
```

Comment:
This runs only once. If new nodejs processes spawn during the lifetime of gProfiler, we want them to get the DSO loaded as well. I suggest you can `generate_map_for_node_processes()` upon every `snapshot()` call?

Reply:
I see that you added in `snapshot` - there's no need to keep this in `start` now, only in `snapshot` suffices (and clean_up in `stop`)

Reply:
Ping @kwisniewski98 :)

Reply:
On a second thought, it's okay to leave it here I suppose - it will affect immediately before perf starts recording. In `snapshot` it's too late for that